### ISSUE TITLE
EqualsAvoidNull checks for objects other than Strings? 

### ISSUE THREAD

**Issue by srangwal** [checkstyle/checkstyle#2816]
Enabling `EqualsAvoidNull` for checkstyle 6.1.1 we find that it reports an issue at the final return statement in the code below

```java 
 public boolean equals(Vector y) {
    if (y == null) {
      return false;
    }
   ... 
   ....
    return y.equals(this);
}
``` 
Here `Vector` is an interface. It is not possible to change the statement to `return this.equals(y)` since it would lead to an infinite call (stackoverflow)

* Does `EqualsAvoidNull` checks for objects other than `String` or this is bug? 
* If so, it looks like the issue reported above is not correct. Is there a way to prevent checkstyle from doing checks on objects other than `String`?

**Comment by romani** [comment]
for future please read: http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug

http://checkstyle.sourceforge.net/config_coding.html#EqualsAvoidNull 
do validation only for String in latest version. It was fixed in some recent releases.

you use 6.1.1 , now we have 6.14.1. You are missing hundreds of bug fixes, see http://checkstyle.sourceforge.net/releasenotes.html.

**Comment by srangwal** [comment]
Thanks. I verified that the problem has been resolved in the current version (6.14.1).
